aulas/modulo5/bloco14/projeto-todo/src/index.ts

```diff
@@ -0,0 +1,260 @@
+        }
+
+        const users = await connection('toDoListUser')
+            .select('id', 'nickname')
+            .where({ id: id })
+
+        if (Object.entries(users).length === 0) {
+            ErrorCode = 422
+            throw new Error("Id não encontrado na database.")
+        }
+
+        res.status(200).send(users)
+    }
+    catch (err: any) {
+        res.status(ErrorCode).end(err.message)
+    }
+})
+
+// EDIT USER BY ID
+app.put('/user/edit/:id', async (req: Request, res: Response) => {
+    let ErrorCode = 500
+    try {
+        const id = Number(req.params.id)
+
+        if (!id) {
+            ErrorCode = 400
+            throw new Error("Parâmetro 'id' não enviado.")
+        }
+
+        const { name, nickname, email } = req.body
+
+        if (name === "" || nickname === "" || email === "") {
+            ErrorCode = 400
+            throw new Error("Parâmetros enviados no body não podem ser vazios.")
+        }
+
+        await EditUser(id, name, nickname, email)
+        res.status(200).send(`Usuário de id:${id} editado com sucesso.`)
+    }
+    catch (err: any) {
+        res.status(ErrorCode).end(err.message)
+    }
+})
+
+// CREATE TODO TASK
+app.post('/task', async (req: Request, res: Response) => {
+    let ErrorCode = 500
+    try {
+        const { task, desc, status, limitDate, creatorId } = req.body
+        if (!task || !desc || !status || !limitDate || !creatorId) {
+            ErrorCode = 400
+            throw new Error("Parâmetro necessário não enviado.")
+        }
+        if (task === "" || desc === "" || limitDate === "" || creatorId === "") {
+            ErrorCode = 400
+            throw new Error("Parâmetros enviados no body não podem ser vazios")
+        }
+
+        const date_regex = /^(?:(?:31(\/|-|\.)(?:0?[13578]|1[02]))\1|(?:(?:29|30)(\/|-|\.)(?:0?[13-9]|1[0-2])\2))(?:(?:1[6-9]|[2-9]\d)?\d{2})$|^(?:29(\/|-|\.)0?2\3(?:(?:(?:1[6-9]|[2-9]\d)?(?:0[48]|[2468][048]|[13579][26])|(?:(?:16|[2468][048]|[3579][26])00))))$|^(?:0?[1-9]|1\d|2[0-8])(\/|-|\.)(?:(?:0?[1-9])|(?:1[0-2]))\4(?:(?:1[6-9]|[2-9]\d)?\d{2})$/
+        if (!(date_regex.test(limitDate))) {
+            ErrorCode = 422
+            throw new Error("Formato de data errado. DD/MM/YYYY")
+        }
+
+        const checkUserId = await connection('toDoListUser')
+            .select()
+            .where({ id: creatorId })
+        if (Object.entries(checkUserId).length === 0) {
+            ErrorCode = 422
+            throw new Error("Id do criador não encontrado na database.")
+        }
+
+        await CreateTask(task, desc, status, limitDate, creatorId)
+
+        res.status(200).send("Tarefa criada com sucesso.")
+    }
+    catch (err: any) {
+        res.status(ErrorCode).end(err.message)
+    }
+})
+
+// GET TODO TASK BY ID
+app.get('/task/:id', async (req: Request, res: Response) => {
+    let ErrorCode = 500
+    try {
+        const id = Number(req.params.id)
+
+        if (!id) {
+            ErrorCode = 400
```

Comment:
Essa validação acaba não funcionando. Por ser params o express já espera o recebimento dessa informação no path! Se não for passado nada ele acusa erro.